src/SnD.Sdk/Storage/Amazon/AmazonBlobStorageService.cs

```diff
@@ -106,15 +124,63 @@ public IEnumerable<StoredBlob> ListBlobsAsEnumerable(string blobPath)
             var response = this.client.ListObjectsV2Async(request).GetAwaiter().GetResult();
             foreach (var s3Object in response.S3Objects)
             {
-                yield return this.MapToStoredBlob(s3Object);
+                yield return MapToStoredBlob(s3Object);
             }
             request.ContinuationToken = response.NextContinuationToken;
         }
         while (request.ContinuationToken != null);
     }
 
+    /// <inheritdoc/>
+    [ExcludeFromCodeCoverage(Justification = "Trivial")]
+    public Task<T> GetBlobContentAsync<T>(string blobPath, string blobName, Func<BinaryData, T> deserializer)
+    {
+        var path = $"{blobPath}/{blobName}".AsAmazonS3Path();
+        var request = new GetObjectRequest
+        {
+            BucketName = path.Bucket,
+            Key = path.ObjectKey
+        };
+        return this.client
+            .GetObjectAsync(request)
+            .TryMap(c =>
+            {
+                using var memoryStream = new MemoryStream();
+                c.ResponseStream.CopyTo(memoryStream);
```

Comment:
Should we use`CopyToAsync` instead?

Reply:
Discussed by voice: if we use `CopyToAsync` here, we will have a hanging task here.